Python/Product/Analysis/AnalysisSet.cs

```diff
@@ -455,6 +455,69 @@ public static class AnalysisSet {
             return set;
         }
 
+        /// <summary>
+        /// Splits values in the set according to the predicate. Returns true
+        /// if there are any values in <paramref name="trueSet"/> on return.
+        /// </summary>
+        public static bool Split(this IAnalysisSet set, Func<AnalysisValue, bool> predicate, out IAnalysisSet trueSet, out IAnalysisSet falseSet) {
```

Comment:
It seems like we have 2 classes for public IAnalysisSet extension methods. They should probably be all moved to AnalysisSetExtensions.cs.

Reply:
The difference is that this one has set methods, while the other broadcasts the methods over the values. That isn't entirely clear, so I'll add internal documentation to the class to clarify.

Reply:
Actually, I think it is clear enough from the class comments already. I'll rename the other class to "AnalysisValueSetExtensions"